storage/tianmu/core/multi_index.h

```diff
@@ -157,33 +176,35 @@ class MultiIndex {
                                                                // groups touched by the set of
                                                                // dimensions
 
-  int no_dimensions;
-  int64_t *dim_size;   // the size of a dimension
-  uint64_t no_tuples;  // actual number of tuples (also in case of virtual
-                       // index); should be updated in any change of index
-  uint32_t p_power;
-  bool no_tuples_too_big;             // this flag is set if a virtual number of tuples
-                                      // exceeds 2^64
-  std::vector<bool> can_be_distinct;  // true if the dimension contain only one copy of
-                                      // original rows, false if we cannot guarantee this
-  std::vector<bool> used_in_output;   // true if given dimension is used for
-                                      // generation of output columns
+  // Some technical functions
+  void MultiplyNoTuples(uint64_t factor);  // the same as "num_of_tuples*=factor", but set
+                                           // has_tuples_too_big whenever needed
 
   // DimensionGroup stuff
   void FillGroupForDim();
-  std::vector<DimensionGroup *> dim_groups;  // all active dimension groups
-  DimensionGroup **group_for_dim;            // pointers to elements of dim_groups, for
-                                             // faster dimension identification
-  int *group_num_for_dim;                    // an element number of dim_groups, for faster
-                                             // dimension identification
 
-  // Some technical functions
-  void MultiplyNoTuples(uint64_t factor);  // the same as "no_tuples*=factor", but set
-                                           // no_tuples_too_big whenever needed
+  uint32_t num_of_dimensions_;
+  int64_t *dim_size_;       // the size of a dimension
+  uint64_t num_of_tuples_;  // actual number of tuples (also in case of virtual index); should be updated in any change
```

Comment:
Should we initialize the member variable in the place where be declared，or in the place of constructure?